Reaper closes connection but doesn't recover

**Expected Behaviour**
For testcontainers to recover internal dependent connections without altering the running system

**Actual Behaviour**
testcontainers looses connection with Reaper, and consecuentally fails running tests

**Testcontainer Logs**

Things seem fine...
```
2025-05-30T16:08:39.5730143Z 2025-05-30T16:08:39.571Z testcontainers [DEBUG] Checking container runtime strategy "TestcontainersHostStrategy"...
2025-05-30T16:08:39.5879332Z 2025-05-30T16:08:39.587Z testcontainers [DEBUG] Container runtime strategy "TestcontainersHostStrategy" is not applicable
2025-05-30T16:08:39.5880515Z 2025-05-30T16:08:39.587Z testcontainers [DEBUG] Checking container runtime strategy "ConfigurationStrategy"...
2025-05-30T16:08:39.5881403Z 2025-05-30T16:08:39.587Z testcontainers [DEBUG] Container runtime strategy "ConfigurationStrategy" is not applicable
2025-05-30T16:08:39.5882369Z 2025-05-30T16:08:39.587Z testcontainers [DEBUG] Checking container runtime strategy "UnixSocketStrategy"...
2025-05-30T16:08:39.5890281Z 2025-05-30T16:08:39.587Z testcontainers [TRACE] Fetching Docker info...
2025-05-30T16:08:39.5965804Z (node:11521) ExperimentalWarning: VM Modules is an experimental feature and might change at any time
2025-05-30T16:08:39.5966910Z (Use `node --trace-warnings ...` to show where the warning was created)
2025-05-30T16:08:39.6636105Z 2025-05-30T16:08:39.662Z testcontainers [TRACE] Fetching remote container runtime socket path...
2025-05-30T16:08:39.6636738Z 2025-05-30T16:08:39.662Z testcontainers [TRACE] Resolving host...
2025-05-30T16:08:39.6637218Z 2025-05-30T16:08:39.662Z testcontainers [DEBUG] Checking gateway for Docker host...
2025-05-30T16:08:39.6678247Z 2025-05-30T16:08:39.667Z testcontainers [TRACE] Fetching Compose info...
2025-05-30T16:08:39.9190230Z 2025-05-30T16:08:39.918Z testcontainers [DEBUG] Checking container runtime strategy "TestcontainersHostStrategy"...
2025-05-30T16:08:39.9199737Z 2025-05-30T16:08:39.919Z testcontainers [DEBUG] Container runtime strategy "TestcontainersHostStrategy" is not applicable
2025-05-30T16:08:39.9210367Z 2025-05-30T16:08:39.920Z testcontainers [DEBUG] Checking container runtime strategy "ConfigurationStrategy"...
2025-05-30T16:08:39.9219927Z 2025-05-30T16:08:39.921Z testcontainers [DEBUG] Container runtime strategy "ConfigurationStrategy" is not applicable
2025-05-30T16:08:39.9232971Z 2025-05-30T16:08:39.922Z testcontainers [DEBUG] Checking container runtime strategy "UnixSocketStrategy"...
2025-05-30T16:08:39.9249425Z 2025-05-30T16:08:39.924Z testcontainers [TRACE] Fetching Docker info...
2025-05-30T16:08:39.9351671Z (node:11539) ExperimentalWarning: VM Modules is an experimental feature and might change at any time
2025-05-30T16:08:39.9353609Z (Use `node --trace-warnings ...` to show where the warning was created)
2025-05-30T16:08:39.9652765Z 2025-05-30T16:08:39.961Z testcontainers [TRACE] Fetching remote container runtime socket path...
2025-05-30T16:08:39.9653654Z 2025-05-30T16:08:39.961Z testcontainers [TRACE] Resolving host...
2025-05-30T16:08:39.9654406Z 2025-05-30T16:08:39.961Z testcontainers [DEBUG] Checking gateway for Docker host...
2025-05-30T16:08:39.9661628Z 2025-05-30T16:08:39.965Z testcontainers [TRACE] Fetching Compose info...
2025-05-30T16:08:40.0766373Z 2025-05-30T16:08:40.074Z testcontainers [DEBUG] Checking container runtime strategy "TestcontainersHostStrategy"...
2025-05-30T16:08:40.0767559Z 2025-05-30T16:08:40.075Z testcontainers [DEBUG] Container runtime strategy "TestcontainersHostStrategy" is not applicable
2025-05-30T16:08:40.0768480Z 2025-05-30T16:08:40.075Z testcontainers [DEBUG] Checking container runtime strategy "ConfigurationStrategy"...
2025-05-30T16:08:40.0769361Z 2025-05-30T16:08:40.075Z testcontainers [DEBUG] Container runtime strategy "ConfigurationStrategy" is not applicable
2025-05-30T16:08:40.0770516Z 2025-05-30T16:08:40.075Z testcontainers [DEBUG] Checking container runtime strategy "UnixSocketStrategy"...
2025-05-30T16:08:40.0771165Z 2025-05-30T16:08:40.076Z testcontainers [TRACE] Fetching Docker info...
2025-05-30T16:08:40.0847386Z (node:11512) ExperimentalWarning: VM Modules is an experimental feature and might change at any time
2025-05-30T16:08:40.0848929Z (Use `node --trace-warnings ...` to show where the warning was created)
2025-05-30T16:08:40.1366673Z 2025-05-30T16:08:40.135Z testcontainers [DEBUG] Checking container runtime strategy "TestcontainersHostStrategy"...
2025-05-30T16:08:40.1378394Z 2025-05-30T16:08:40.136Z testcontainers [DEBUG] Container runtime strategy "TestcontainersHostStrategy" is not applicable
2025-05-30T16:08:40.1379534Z 2025-05-30T16:08:40.136Z testcontainers [DEBUG] Checking container runtime strategy "ConfigurationStrategy"...
2025-05-30T16:08:40.1380562Z 2025-05-30T16:08:40.137Z testcontainers [DEBUG] Container runtime strategy "ConfigurationStrategy" is not applicable
2025-05-30T16:08:40.1381358Z 2025-05-30T16:08:40.137Z testcontainers [DEBUG] Checking container runtime strategy "UnixSocketStrategy"...
2025-05-30T16:08:40.1381949Z 2025-05-30T16:08:40.137Z testcontainers [TRACE] Fetching Docker info...
2025-05-30T16:08:40.1460737Z (node:11511) ExperimentalWarning: VM Modules is an experimental feature and might change at any time
2025-05-30T16:08:40.1461841Z (Use `node --trace-warnings ...` to show where the warning was created)
2025-05-30T16:08:40.1603892Z 2025-05-30T16:08:40.157Z testcontainers [TRACE] Fetching remote container runtime socket path...
2025-05-30T16:08:40.1604502Z 2025-05-30T16:08:40.157Z testcontainers [TRACE] Resolving host...
2025-05-30T16:08:40.1604984Z 2025-05-30T16:08:40.157Z testcontainers [DEBUG] Checking gateway for Docker host...
2025-05-30T16:08:40.1605758Z 2025-05-30T16:08:40.157Z testcontainers [TRACE] Fetching remote container runtime socket path...
2025-05-30T16:08:40.1606379Z 2025-05-30T16:08:40.157Z testcontainers [TRACE] Resolving host...
2025-05-30T16:08:40.1606940Z 2025-05-30T16:08:40.157Z testcontainers [DEBUG] Checking gateway for Docker host...
2025-05-30T16:08:40.1627351Z 2025-05-30T16:08:40.162Z testcontainers [TRACE] Fetching Compose info...
2025-05-30T16:08:40.1698647Z 2025-05-30T16:08:40.164Z testcontainers [TRACE] Fetching Compose info...
2025-05-30T16:08:40.4417024Z 2025-05-30T16:08:40.441Z testcontainers [TRACE] Looking up host IPs...
2025-05-30T16:08:40.4421893Z 2025-05-30T16:08:40.441Z testcontainers [TRACE] Initialising clients...
2025-05-30T16:08:40.4422366Z 2025-05-30T16:08:40.441Z testcontainers [TRACE] Container runtime info:
2025-05-30T16:08:40.4422897Z {
2025-05-30T16:08:40.4423071Z   "node": {
2025-05-30T16:08:40.4423322Z     "version": "v20.19.2",
2025-05-30T16:08:40.4423822Z     "architecture": "arm64",
2025-05-30T16:08:40.4424237Z     "platform": "linux"
2025-05-30T16:08:40.4427408Z   },
2025-05-30T16:08:40.4427687Z   "containerRuntime": {
2025-05-30T16:08:40.4428060Z     "host": "172.17.0.1",
2025-05-30T16:08:40.4428574Z     "hostIps": [
2025-05-30T16:08:40.4428803Z       {
2025-05-30T16:08:40.4428979Z         "address": "172.17.0.1",
2025-05-30T16:08:40.4429217Z         "family": 4
2025-05-30T16:08:40.4429398Z       }
2025-05-30T16:08:40.4429682Z     ],
2025-05-30T16:08:40.4429889Z     "remoteSocketPath": "/var/run/docker.sock",
2025-05-30T16:08:40.4430268Z     "indexServerAddress": "https://index.docker.io/v1/",
2025-05-30T16:08:40.4430607Z     "serverVersion": "28.1.1",
2025-05-30T16:08:40.4430907Z     "operatingSystem": "Ubuntu 22.04.5 LTS (containerized)",
2025-05-30T16:08:40.4431245Z     "operatingSystemType": "linux",
2025-05-30T16:08:40.4431506Z     "architecture": "aarch64",
2025-05-30T16:08:40.4431738Z     "cpus": 8,
2025-05-30T16:08:40.4431923Z     "memory": 16577662976,
2025-05-30T16:08:40.4432137Z     "runtimes": [
2025-05-30T16:08:40.4432339Z       "io.containerd.runc.v2",
2025-05-30T16:08:40.4432560Z       "runc"
2025-05-30T16:08:40.4432728Z     ],
2025-05-30T16:08:40.4432889Z     "labels": []
2025-05-30T16:08:40.4433065Z   },
2025-05-30T16:08:40.4433222Z   "compose": {
2025-05-30T16:08:40.4433405Z     "version": "2.35.1",
2025-05-30T16:08:40.4433623Z     "compatability": "v2"
2025-05-30T16:08:40.4433831Z   }
2025-05-30T16:08:40.4433992Z }
```

Then a "Connection is closed"...:
```
2025-05-30T16:11:30.6918155Z 2025-05-30T16:11:30.691Z testcontainers [DEBUG] Checking container runtime strategy "TestcontainersHostStrategy"...
2025-05-30T16:11:30.6927150Z 2025-05-30T16:11:30.691Z testcontainers [DEBUG] Container runtime strategy "TestcontainersHostStrategy" is not applicable
2025-05-30T16:11:30.6928241Z 2025-05-30T16:11:30.691Z testcontainers [DEBUG] Checking container runtime strategy "ConfigurationStrategy"...
2025-05-30T16:11:30.6929225Z 2025-05-30T16:11:30.691Z testcontainers [DEBUG] Container runtime strategy "ConfigurationStrategy" is not applicable
2025-05-30T16:11:30.6930101Z 2025-05-30T16:11:30.691Z testcontainers [DEBUG] Checking container runtime strategy "UnixSocketStrategy"...
2025-05-30T16:11:30.6930846Z 2025-05-30T16:11:30.692Z testcontainers [TRACE] Fetching Docker info...
2025-05-30T16:11:30.7316834Z 2025-05-30T16:11:30.730Z testcontainers [TRACE] Fetching remote container runtime socket path...
2025-05-30T16:11:30.7317741Z 2025-05-30T16:11:30.730Z testcontainers [TRACE] Resolving host...
2025-05-30T16:11:30.7318417Z 2025-05-30T16:11:30.730Z testcontainers [DEBUG] Checking gateway for Docker host...
2025-05-30T16:11:30.7366628Z 2025-05-30T16:11:30.734Z testcontainers [TRACE] Fetching Compose info...
2025-05-30T16:11:30.7744593Z 2025-05-30T16:11:30.773Z testcontainers [TRACE] Looking up host IPs...
2025-05-30T16:11:30.7745825Z 2025-05-30T16:11:30.774Z testcontainers [TRACE] Initialising clients...
2025-05-30T16:11:30.7746760Z 2025-05-30T16:11:30.774Z testcontainers [TRACE] Container runtime info:
2025-05-30T16:11:30.7747549Z {
2025-05-30T16:11:30.7748523Z   "node": {
2025-05-30T16:11:30.7749037Z     "version": "v20.19.2",
2025-05-30T16:11:30.7749509Z     "architecture": "arm64",
2025-05-30T16:11:30.7750154Z     "platform": "linux"
2025-05-30T16:11:30.7750522Z   },
2025-05-30T16:11:30.7752295Z   "containerRuntime": {
2025-05-30T16:11:30.7752767Z     "host": "172.17.0.1",
2025-05-30T16:11:30.7753085Z     "hostIps": [
2025-05-30T16:11:30.7753553Z       {
2025-05-30T16:11:30.7753929Z         "address": "172.17.0.1",
2025-05-30T16:11:30.7756147Z         "family": 4
2025-05-30T16:11:30.7760130Z       }
2025-05-30T16:11:30.7760935Z     ],
2025-05-30T16:11:30.7761621Z     "remoteSocketPath": "/var/run/docker.sock",
2025-05-30T16:11:30.7762464Z     "indexServerAddress": "https://index.docker.io/v1/",
2025-05-30T16:11:30.7763111Z     "serverVersion": "28.1.1",
2025-05-30T16:11:30.7763549Z     "operatingSystem": "Ubuntu 22.04.5 LTS (containerized)",
2025-05-30T16:11:30.7847278Z     "operatingSystemType": "linux",
2025-05-30T16:11:30.7847643Z     "architecture": "aarch64",
2025-05-30T16:11:30.7847951Z     "cpus": 8,
2025-05-30T16:11:30.7848136Z     "memory": 16577662976,
2025-05-30T16:11:30.7848345Z     "runtimes": [
2025-05-30T16:11:30.7907656Z       "io.containerd.runc.v2",
2025-05-30T16:11:30.7908232Z       "runc"
2025-05-30T16:11:30.7908588Z     ],
2025-05-30T16:11:30.7908791Z     "labels": []
2025-05-30T16:11:30.7908972Z   },
2025-05-30T16:11:30.7909130Z   "compose": {
2025-05-30T16:11:30.7909415Z     "version": "2.35.1",
2025-05-30T16:11:30.7909756Z     "compatability": "v2"
2025-05-30T16:11:30.7910047Z   }
2025-05-30T16:11:30.7910414Z }
2025-05-30T16:11:30.7911050Z 2025-05-30T16:11:30.774Z testcontainers [DEBUG] Container runtime strategy "UnixSocketStrategy" works
2025-05-30T16:11:30.7911837Z 2025-05-30T16:11:30.774Z testcontainers [DEBUG] Checking if image exists "mysql:8.0.39"...
2025-05-30T16:11:30.7914083Z 2025-05-30T16:11:30.782Z testcontainers [DEBUG] Checked if image exists "mysql:8.0.39"
2025-05-30T16:11:30.7915146Z 2025-05-30T16:11:30.782Z testcontainers [DEBUG] Pulling image "mysql:8.0.39"...
2025-05-30T16:11:30.7915940Z 2025-05-30T16:11:30.783Z testcontainers [DEBUG] No registry auth locator found for registry "https://index.docker.io/v1/"
2025-05-30T16:11:30.8071822Z 2025-05-30T16:11:30.806Z testcontainers [DEBUG] [927936aca11c] Started container
2025-05-30T16:11:30.8072548Z 2025-05-30T16:11:30.806Z testcontainers [INFO] [927936aca11c] Started container for image "amazon/dynamodb-local:1.20.0"
2025-05-30T16:11:30.8073226Z 2025-05-30T16:11:30.806Z testcontainers [DEBUG] [927936aca11c] Inspecting container...
2025-05-30T16:11:30.8104902Z 2025-05-30T16:11:30.809Z testcontainers [DEBUG] [927936aca11c] Inspected container
2025-05-30T16:11:30.8105991Z 2025-05-30T16:11:30.809Z testcontainers [DEBUG] [927936aca11c] Fetching container logs...
2025-05-30T16:11:30.8127986Z 2025-05-30T16:11:30.812Z testcontainers [DEBUG] [927936aca11c] Demuxing stream...
2025-05-30T16:11:30.8135371Z 2025-05-30T16:11:30.812Z testcontainers [DEBUG] [927936aca11c] Demuxed stream
2025-05-30T16:11:30.8146774Z 2025-05-30T16:11:30.814Z testcontainers [DEBUG] [927936aca11c] Fetched container logs
2025-05-30T16:11:30.8147621Z 2025-05-30T16:11:30.814Z testcontainers [DEBUG] [927936aca11c] Waiting for container to be ready...
2025-05-30T16:11:30.8151024Z 2025-05-30T16:11:30.814Z testcontainers [DEBUG] [927936aca11c] Waiting for host port 32818...
2025-05-30T16:11:30.8159289Z 2025-05-30T16:11:30.815Z testcontainers [DEBUG] [927936aca11c] Waiting for internal port 8000...
2025-05-30T16:11:30.8238146Z 2025-05-30T16:11:30.822Z testcontainers [DEBUG] [927936aca11c] Host port 32818 ready
2025-05-30T16:11:30.8262751Z 2025-05-30T16:11:30.823Z testcontainers [DEBUG] [927936aca11c] Host port wait strategy complete
2025-05-30T16:11:30.8503984Z 2025-05-30T16:11:30.849Z testcontainers [WARN] [6f06abd1da49] Connection to Reaper closed
2025-05-30T16:11:30.8541173Z 2025-05-30T16:11:30.849Z testcontainers [WARN] [6f06abd1da49] Connection to Reaper closed
```

Then the world ends... 👎🏽 
```
2025-05-30T16:11:31.3324352Z 2025-05-30T16:11:31.330Z testcontainers [TRACE] Looking up host IPs...
2025-05-30T16:11:31.3325141Z 2025-05-30T16:11:31.330Z testcontainers [TRACE] Initialising clients...
2025-05-30T16:11:31.3325823Z 2025-05-30T16:11:31.330Z testcontainers [TRACE] Container runtime info:
2025-05-30T16:11:31.3326359Z {
2025-05-30T16:11:31.3327082Z   "node": {
2025-05-30T16:11:31.3327374Z     "version": "v20.19.2",
2025-05-30T16:11:31.3327727Z     "architecture": "arm64",
2025-05-30T16:11:31.3328097Z     "platform": "linux"
2025-05-30T16:11:31.3328406Z   },
2025-05-30T16:11:31.3328683Z   "containerRuntime": {
2025-05-30T16:11:31.3328966Z     "host": "172.17.0.1",
2025-05-30T16:11:31.3329226Z     "hostIps": [
2025-05-30T16:11:31.3329460Z       {
2025-05-30T16:11:31.3329677Z         "address": "172.17.0.1",
2025-05-30T16:11:31.3329953Z         "family": 4
2025-05-30T16:11:31.3330187Z       }
2025-05-30T16:11:31.3330394Z     ],
2025-05-30T16:11:31.3330669Z     "remoteSocketPath": "/var/run/docker.sock",
2025-05-30T16:11:31.3331103Z     "indexServerAddress": "https://index.docker.io/v1/",
2025-05-30T16:11:31.3331505Z     "serverVersion": "28.1.1",
2025-05-30T16:11:31.3331872Z     "operatingSystem": "Ubuntu 22.04.5 LTS (containerized)",
2025-05-30T16:11:31.3332280Z     "operatingSystemType": "linux",
2025-05-30T16:11:31.3332599Z     "architecture": "aarch64",
2025-05-30T16:11:31.3332874Z     "cpus": 8,
2025-05-30T16:11:31.3333112Z     "memory": 16577662976,
2025-05-30T16:11:31.3333370Z     "runtimes": [
2025-05-30T16:11:31.3333625Z       "io.containerd.runc.v2",
2025-05-30T16:11:31.3333899Z       "runc"
2025-05-30T16:11:31.3334120Z     ],
2025-05-30T16:11:31.3334327Z     "labels": []
2025-05-30T16:11:31.3334550Z   },
2025-05-30T16:11:31.3334760Z   "compose": {
2025-05-30T16:11:31.3334990Z     "version": "2.35.1",
2025-05-30T16:11:31.3335259Z     "compatability": "v2"
2025-05-30T16:11:31.3335516Z   }
2025-05-30T16:11:31.3335724Z }
2025-05-30T16:11:31.3336160Z 2025-05-30T16:11:31.330Z testcontainers [DEBUG] Container runtime strategy "UnixSocketStrategy" works
2025-05-30T16:11:31.3337266Z 2025-05-30T16:11:31.331Z testcontainers [DEBUG] Checking if image exists "localstack/localstack:3.0.2"...
2025-05-30T16:11:31.3353861Z 2025-05-30T16:11:31.334Z testcontainers [DEBUG] Checked if image exists "localstack/localstack:3.0.2"
2025-05-30T16:11:31.3356582Z 2025-05-30T16:11:31.335Z testcontainers [DEBUG] Image "localstack/localstack:3.0.2" already exists
2025-05-30T16:11:31.3530870Z 2025-05-30T16:11:31.352Z testcontainers [DEBUG] Acquiring lock file "/tmp/testcontainers-node.lock"...
2025-05-30T16:11:31.3544898Z 2025-05-30T16:11:31.353Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":335,"total":335},"progress":"[==================================================\u003e]     335B/335B","id":"f6d5909e94cd"}
2025-05-30T16:11:31.3546000Z 2025-05-30T16:11:31.353Z testcontainers:pull [mysql:8.0.39] {"status":"Verifying Checksum","progressDetail":{},"id":"f6d5909e94cd"}
2025-05-30T16:11:31.3546853Z 2025-05-30T16:11:31.354Z testcontainers:pull [mysql:8.0.39] {"status":"Download complete","progressDetail":{},"id":"f6d5909e94cd"}
2025-05-30T16:11:31.3566688Z 2025-05-30T16:11:31.355Z testcontainers [DEBUG] Acquired lock file "/tmp/testcontainers-node.lock"
2025-05-30T16:11:31.3574177Z 2025-05-30T16:11:31.355Z testcontainers [DEBUG] Listing containers...
2025-05-30T16:11:31.3603912Z 2025-05-30T16:11:31.359Z testcontainers [DEBUG] Listed containers
2025-05-30T16:11:31.3626048Z 2025-05-30T16:11:31.362Z testcontainers [DEBUG] Reusing existing Reaper for session "ef498acb902b"...
2025-05-30T16:11:31.3632011Z 2025-05-30T16:11:31.362Z testcontainers [DEBUG] [6f06abd1da49] Connecting to Reaper (attempt 1) on "172.17.0.1:32768"...
2025-05-30T16:11:31.3659676Z 2025-05-30T16:11:31.365Z testcontainers [ERROR] Reaper 6f06abd1da49576a9a98c3ae673165b0881f828cfa719eef3ff57c049472e50e socket error: Error: connect ECONNREFUSED 172.17.0.1:32768
2025-05-30T16:11:31.3661291Z 2025-05-30T16:11:31.365Z testcontainers [ERROR] [6f06abd1da49] Connection to Reaper closed with error
2025-05-30T16:11:31.3737055Z 2025-05-30T16:11:31.371Z testcontainers [TRACE] Fetching remote container runtime socket path...
2025-05-30T16:11:31.3737866Z 2025-05-30T16:11:31.371Z testcontainers [TRACE] Resolving host...
2025-05-30T16:11:31.3738452Z 2025-05-30T16:11:31.371Z testcontainers [DEBUG] Checking gateway for Docker host...
2025-05-30T16:11:31.3765448Z 2025-05-30T16:11:31.376Z testcontainers [TRACE] Fetching Compose info...
2025-05-30T16:11:31.3780007Z 2025-05-30T16:11:31.377Z testcontainers [DEBUG] [0aeea7f9867f] Started container
2025-05-30T16:11:31.3780942Z 2025-05-30T16:11:31.377Z testcontainers [INFO] [0aeea7f9867f] Started container for image "wiremock/wiremock:3.5.4-1"
2025-05-30T16:11:31.3781812Z 2025-05-30T16:11:31.377Z testcontainers [DEBUG] [0aeea7f9867f] Inspecting container...
2025-05-30T16:11:31.3795734Z 2025-05-30T16:11:31.378Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":26439828,"total":47914583},"progress":"[===========================\u003e                       ]  26.44MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:31.3854016Z 2025-05-30T16:11:31.382Z testcontainers [DEBUG] [0aeea7f9867f] Inspected container
2025-05-30T16:11:31.3855541Z 2025-05-30T16:11:31.383Z testcontainers [DEBUG] [0aeea7f9867f] Fetching container logs...
2025-05-30T16:11:31.3857639Z 2025-05-30T16:11:31.385Z testcontainers [DEBUG] [0aeea7f9867f] Demuxing stream...
2025-05-30T16:11:31.3863048Z 2025-05-30T16:11:31.386Z testcontainers [DEBUG] [0aeea7f9867f] Demuxed stream
2025-05-30T16:11:31.3865014Z 2025-05-30T16:11:31.386Z testcontainers [DEBUG] [0aeea7f9867f] Fetched container logs
2025-05-30T16:11:31.3866481Z 2025-05-30T16:11:31.386Z testcontainers [DEBUG] [0aeea7f9867f] Waiting for container to be ready...
2025-05-30T16:11:31.3868304Z 2025-05-30T16:11:31.386Z testcontainers [DEBUG] [0aeea7f9867f] Waiting for host port 32819...
2025-05-30T16:11:31.3872512Z 2025-05-30T16:11:31.387Z testcontainers [DEBUG] [0aeea7f9867f] Waiting for internal port 8080...
2025-05-30T16:11:31.3934152Z 2025-05-30T16:11:31.391Z testcontainers [DEBUG] [0aeea7f9867f] Host port 32819 ready
2025-05-30T16:11:31.3935213Z 2025-05-30T16:11:31.391Z testcontainers [DEBUG] [0aeea7f9867f] Host port wait strategy complete
2025-05-30T16:11:31.4369194Z 2025-05-30T16:11:31.436Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":316,"total":316},"progress":"[==================================================\u003e]     316B/316B","id":"5355ea40a182"}
2025-05-30T16:11:31.4373700Z 2025-05-30T16:11:31.436Z testcontainers:pull [mysql:8.0.39] {"status":"Verifying Checksum","progressDetail":{},"id":"5355ea40a182"}
2025-05-30T16:11:31.4374820Z 2025-05-30T16:11:31.436Z testcontainers:pull [mysql:8.0.39] {"status":"Download complete","progressDetail":{},"id":"5355ea40a182"}
2025-05-30T16:11:31.5028369Z 2025-05-30T16:11:31.502Z testcontainers [TRACE] [0aeea7f9867f] Port check result exit code 127: /bin/sh: 1: nc: not found
2025-05-30T16:11:31.5029533Z 2025-05-30T16:11:31.502Z testcontainers [TRACE] [0aeea7f9867f] Port check result exit code 1: /bin/bash: connect: Connection refused
2025-05-30T16:11:31.5030561Z /bin/bash: line 1: /dev/tcp/localhost/8080: Connection refused
2025-05-30T16:11:31.5132947Z 2025-05-30T16:11:31.512Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":32813444,"total":47914583},"progress":"[==================================\u003e                ]  32.81MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:31.5219924Z PASS __tests__/sql.test.ts
2025-05-30T16:11:31.5347291Z 2025-05-30T16:11:31.534Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":480988,"total":48048247},"progress":"[\u003e                                                  ]    481kB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:31.6117932Z PASS __tests__/adapters/driver/HeadCertificateHandler.test.ts
2025-05-30T16:11:31.6745319Z 2025-05-30T16:11:31.673Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":538095,"total":68412672},"progress":"[\u003e                                                  ]  538.1kB/68.41MB","id":"2a489f3632fa"}
2025-05-30T16:11:31.6888684Z 2025-05-30T16:11:31.687Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":33789596,"total":47914583},"progress":"[===================================\u003e               ]  33.79MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:31.7495006Z 2025-05-30T16:11:31.749Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":18071444,"total":48048247},"progress":"[==================\u003e                                ]  18.07MB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:31.7796728Z 2025-05-30T16:11:31.779Z testcontainers:pull [mysql:8.0.39] {"status":"Verifying Checksum","progressDetail":{},"id":"8b4274ea61c5"}
2025-05-30T16:11:31.7797907Z 2025-05-30T16:11:31.779Z testcontainers:pull [mysql:8.0.39] {"status":"Download complete","progressDetail":{},"id":"8b4274ea61c5"}
2025-05-30T16:11:31.8372833Z 2025-05-30T16:11:31.836Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":25914596,"total":68412672},"progress":"[==================\u003e                                ]  25.91MB/68.41MB","id":"2a489f3632fa"}
2025-05-30T16:11:31.9068204Z 2025-05-30T16:11:31.904Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":33789596,"total":48048247},"progress":"[===================================\u003e               ]  33.79MB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:31.9243547Z 2025-05-30T16:11:31.919Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":5324,"total":5324},"progress":"[==================================================\u003e]  5.324kB/5.324kB","id":"efd64d077766"}
2025-05-30T16:11:31.9244814Z 2025-05-30T16:11:31.919Z testcontainers:pull [mysql:8.0.39] {"status":"Verifying Checksum","progressDetail":{},"id":"efd64d077766"}
2025-05-30T16:11:31.9245688Z 2025-05-30T16:11:31.919Z testcontainers:pull [mysql:8.0.39] {"status":"Download complete","progressDetail":{},"id":"efd64d077766"}
2025-05-30T16:11:31.9300844Z 2025-05-30T16:11:31.929Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":491520,"total":47914583},"progress":"[\u003e                                                  ]  491.5kB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:31.9447441Z 2025-05-30T16:11:31.944Z testcontainers:pull [mysql:8.0.39] {"status":"Verifying Checksum","progressDetail":{},"id":"949fd89ec12a"}
2025-05-30T16:11:31.9448619Z 2025-05-30T16:11:31.944Z testcontainers:pull [mysql:8.0.39] {"status":"Download complete","progressDetail":{},"id":"949fd89ec12a"}
2025-05-30T16:11:31.9726553Z 2025-05-30T16:11:31.971Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":26452980,"total":68412672},"progress":"[===================\u003e                               ]  26.45MB/68.41MB","id":"2a489f3632fa"}
2025-05-30T16:11:31.9815114Z 2025-05-30T16:11:31.980Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":122,"total":122},"progress":"[==================================================\u003e]     122B/122B","id":"e6689b530d4e"}
2025-05-30T16:11:31.9945804Z 2025-05-30T16:11:31.981Z testcontainers:pull [mysql:8.0.39] {"status":"Verifying Checksum","progressDetail":{},"id":"e6689b530d4e"}
2025-05-30T16:11:31.9947247Z 2025-05-30T16:11:31.984Z testcontainers:pull [mysql:8.0.39] {"status":"Download complete","progressDetail":{},"id":"e6689b530d4e"}
2025-05-30T16:11:31.9948154Z 2025-05-30T16:11:31.985Z testcontainers [TRACE] Looking up host IPs...
2025-05-30T16:11:31.9948931Z 2025-05-30T16:11:31.986Z testcontainers [TRACE] Initialising clients...
2025-05-30T16:11:31.9949806Z 2025-05-30T16:11:31.986Z testcontainers [TRACE] Container runtime info:
2025-05-30T16:11:31.9950204Z {
2025-05-30T16:11:31.9950686Z   "node": {
2025-05-30T16:11:31.9950872Z     "version": "v20.19.2",
2025-05-30T16:11:31.9951127Z     "architecture": "arm64",
2025-05-30T16:11:31.9951380Z     "platform": "linux"
2025-05-30T16:11:31.9951578Z   },
2025-05-30T16:11:31.9951751Z   "containerRuntime": {
2025-05-30T16:11:31.9951962Z     "host": "172.17.0.1",
2025-05-30T16:11:31.9952171Z     "hostIps": [
2025-05-30T16:11:31.9952352Z       {
2025-05-30T16:11:31.9952523Z         "address": "172.17.0.1",
2025-05-30T16:11:31.9952751Z         "family": 4
2025-05-30T16:11:31.9952933Z       }
2025-05-30T16:11:31.9953087Z     ],
2025-05-30T16:11:31.9953296Z     "remoteSocketPath": "/var/run/docker.sock",
2025-05-30T16:11:31.9953672Z     "indexServerAddress": "https://index.docker.io/v1/",
2025-05-30T16:11:31.9954003Z     "serverVersion": "28.1.1",
2025-05-30T16:11:31.9954316Z     "operatingSystem": "Ubuntu 22.04.5 LTS (containerized)",
2025-05-30T16:11:31.9954656Z     "operatingSystemType": "linux",
2025-05-30T16:11:31.9954922Z     "architecture": "aarch64",
2025-05-30T16:11:31.9955160Z     "cpus": 8,
2025-05-30T16:11:31.9955345Z     "memory": 16577662976,
2025-05-30T16:11:31.9955566Z     "runtimes": [
2025-05-30T16:11:31.9955770Z       "io.containerd.runc.v2",
2025-05-30T16:11:31.9955995Z       "runc"
2025-05-30T16:11:31.9956165Z     ],
2025-05-30T16:11:31.9956327Z     "labels": []
2025-05-30T16:11:31.9956505Z   },
2025-05-30T16:11:31.9956665Z   "compose": {
2025-05-30T16:11:31.9956847Z     "version": "2.35.1",
2025-05-30T16:11:31.9957074Z     "compatability": "v2"
2025-05-30T16:11:31.9957286Z   }
2025-05-30T16:11:31.9957437Z }
2025-05-30T16:11:31.9957821Z 2025-05-30T16:11:31.986Z testcontainers [DEBUG] Container runtime strategy "UnixSocketStrategy" works
2025-05-30T16:11:31.9958507Z 2025-05-30T16:11:31.987Z testcontainers [DEBUG] Checking if image exists "amazon/dynamodb-local:1.20.0"...
2025-05-30T16:11:31.9959178Z 2025-05-30T16:11:31.991Z testcontainers [DEBUG] Checked if image exists "amazon/dynamodb-local:1.20.0"
2025-05-30T16:11:31.9959836Z 2025-05-30T16:11:31.991Z testcontainers [DEBUG] Image "amazon/dynamodb-local:1.20.0" already exists
2025-05-30T16:11:32.0103793Z 2025-05-30T16:11:32.009Z testcontainers [DEBUG] Acquiring lock file "/tmp/testcontainers-node.lock"...
2025-05-30T16:11:32.0442035Z 2025-05-30T16:11:32.043Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":1966080,"total":47914583},"progress":"[==\u003e                                                ]  1.966MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:32.1625865Z 2025-05-30T16:11:32.161Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":6389760,"total":47914583},"progress":"[======\u003e                                            ]   6.39MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:32.1944224Z 2025-05-30T16:11:32.193Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":33935212,"total":68412672},"progress":"[========================\u003e                          ]  33.94MB/68.41MB","id":"2a489f3632fa"}
2025-05-30T16:11:32.2957489Z 2025-05-30T16:11:32.294Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":58958444,"total":68412672},"progress":"[===========================================\u003e       ]  58.96MB/68.41MB","id":"2a489f3632fa"}
2025-05-30T16:11:32.3030667Z 2025-05-30T16:11:32.302Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":7864320,"total":47914583},"progress":"[========\u003e                                          ]  7.864MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:32.3683667Z 2025-05-30T16:11:32.367Z testcontainers [DEBUG] [6f06abd1da49] Connecting to Reaper (attempt 2) on "172.17.0.1:32768"...
2025-05-30T16:11:32.3685053Z 2025-05-30T16:11:32.367Z testcontainers [ERROR] Reaper 6f06abd1da49576a9a98c3ae673165b0881f828cfa719eef3ff57c049472e50e socket error: Error: connect ECONNREFUSED 172.17.0.1:32768
2025-05-30T16:11:32.3686612Z 2025-05-30T16:11:32.367Z testcontainers [ERROR] [6f06abd1da49] Connection to Reaper closed with error
2025-05-30T16:11:32.4055988Z 2025-05-30T16:11:32.404Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":12779520,"total":47914583},"progress":"[=============\u003e                                     ]  12.78MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:32.5076399Z 2025-05-30T16:11:32.507Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":18186240,"total":47914583},"progress":"[==================\u003e                                ]  18.19MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:32.5192753Z 2025-05-30T16:11:32.512Z testcontainers [DEBUG] [927936aca11c] Internal port 8000 ready
2025-05-30T16:11:32.5246894Z 2025-05-30T16:11:32.512Z testcontainers [INFO] [927936aca11c] Container is ready
2025-05-30T16:11:32.5542336Z 2025-05-30T16:11:32.553Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":60020444,"total":68412672},"progress":"[===========================================\u003e       ]  60.02MB/68.41MB","id":"2a489f3632fa"}
2025-05-30T16:11:32.6156921Z 2025-05-30T16:11:32.615Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":22118400,"total":47914583},"progress":"[=======================\u003e                           ]  22.12MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:32.6815656Z 2025-05-30T16:11:32.680Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":66444828,"total":68412672},"progress":"[================================================\u003e  ]  66.44MB/68.41MB","id":"2a489f3632fa"}
2025-05-30T16:11:32.7172675Z 2025-05-30T16:11:32.716Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":25067520,"total":47914583},"progress":"[==========================\u003e                        ]  25.07MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:32.7848852Z 2025-05-30T16:11:32.783Z testcontainers:containers [0aeea7f9867f] 
2025-05-30T16:11:32.7849968Z 2025-05-30T16:11:32.783Z testcontainers:containers [0aeea7f9867f] [0m----------------------------------------------------------------
2025-05-30T16:11:32.7851108Z 2025-05-30T16:11:32.783Z testcontainers:containers [0aeea7f9867f] |               Cloud: https://wiremock.io/cloud               |
2025-05-30T16:11:32.7851883Z 2025-05-30T16:11:32.783Z testcontainers:containers [0aeea7f9867f] |                                                              |
2025-05-30T16:11:32.7852634Z 2025-05-30T16:11:32.783Z testcontainers:containers [0aeea7f9867f] |               Slack: https://slack.wiremock.org              |
2025-05-30T16:11:32.7853415Z 2025-05-30T16:11:32.783Z testcontainers:containers [0aeea7f9867f] ----------------------------------------------------------------
2025-05-30T16:11:32.7854454Z 2025-05-30T16:11:32.783Z testcontainers:pull [mysql:8.0.39] {"status":"Downloading","progressDetail":{"current":67506828,"total":68412672},"progress":"[=================================================\u003e ]  67.51MB/68.41MB","id":"2a489f3632fa"}
2025-05-30T16:11:32.7855930Z 2025-05-30T16:11:32.785Z testcontainers:containers [0aeea7f9867f] 
2025-05-30T16:11:32.7879318Z 2025-05-30T16:11:32.787Z testcontainers:pull [mysql:8.0.39] {"status":"Verifying Checksum","progressDetail":{},"id":"2a489f3632fa"}
2025-05-30T16:11:32.7880210Z 2025-05-30T16:11:32.787Z testcontainers:pull [mysql:8.0.39] {"status":"Download complete","progressDetail":{},"id":"2a489f3632fa"}
2025-05-30T16:11:32.7898151Z 2025-05-30T16:11:32.789Z testcontainers:containers [0aeea7f9867f] version:                      3.5.4
2025-05-30T16:11:32.7898840Z 2025-05-30T16:11:32.789Z testcontainers:containers [0aeea7f9867f] port:                         8080
2025-05-30T16:11:32.7899494Z 2025-05-30T16:11:32.789Z testcontainers:containers [0aeea7f9867f] enable-browser-proxying:      false
2025-05-30T16:11:32.7900150Z 2025-05-30T16:11:32.789Z testcontainers:containers [0aeea7f9867f] disable-banner:               false
2025-05-30T16:11:32.7900798Z 2025-05-30T16:11:32.789Z testcontainers:containers [0aeea7f9867f] no-request-journal:           false
2025-05-30T16:11:32.7901474Z 2025-05-30T16:11:32.789Z testcontainers:containers [0aeea7f9867f] verbose:                      false
2025-05-30T16:11:32.7902292Z 2025-05-30T16:11:32.789Z testcontainers:containers [0aeea7f9867f] 
2025-05-30T16:11:32.7969927Z 2025-05-30T16:11:32.796Z testcontainers:containers [0aeea7f9867f] extensions:                   response-template,webhook
2025-05-30T16:11:32.8289899Z 2025-05-30T16:11:32.827Z testcontainers [DEBUG] [0aeea7f9867f] Internal port 8080 ready
2025-05-30T16:11:32.8317567Z 2025-05-30T16:11:32.827Z testcontainers [INFO] [0aeea7f9867f] Container is ready
2025-05-30T16:11:32.8333980Z 2025-05-30T16:11:32.832Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":28508160,"total":47914583},"progress":"[=============================\u003e                     ]  28.51MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:32.9103928Z PASS __tests__/cursor.test.ts
2025-05-30T16:11:32.9585848Z 2025-05-30T16:11:32.958Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":30965760,"total":47914583},"progress":"[================================\u003e                  ]  30.97MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:33.0745398Z 2025-05-30T16:11:33.074Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":41779200,"total":47914583},"progress":"[===========================================\u003e       ]  41.78MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:33.1171124Z PASS __tests__/ts/BulkAssetValidator.test.ts
2025-05-30T16:11:33.1919630Z 2025-05-30T16:11:33.191Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":43745280,"total":47914583},"progress":"[=============================================\u003e     ]  43.75MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:33.2313684Z PASS __tests__/adapters/driver/GetCertificatePreSignedUrlHandler.test.ts
2025-05-30T16:11:33.3696541Z 2025-05-30T16:11:33.368Z testcontainers [DEBUG] [6f06abd1da49] Connecting to Reaper (attempt 3) on "172.17.0.1:32768"...
2025-05-30T16:11:33.3702991Z 2025-05-30T16:11:33.369Z testcontainers [ERROR] Reaper 6f06abd1da49576a9a98c3ae673165b0881f828cfa719eef3ff57c049472e50e socket error: Error: connect ECONNREFUSED 172.17.0.1:32768
2025-05-30T16:11:33.3704090Z 2025-05-30T16:11:33.369Z testcontainers [ERROR] [6f06abd1da49] Connection to Reaper closed with error
2025-05-30T16:11:33.3861717Z 2025-05-30T16:11:33.385Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":44728320,"total":47914583},"progress":"[==============================================\u003e    ]  44.73MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:33.4941343Z 2025-05-30T16:11:33.493Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":46202880,"total":47914583},"progress":"[================================================\u003e  ]   46.2MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:33.5582545Z 2025-05-30T16:11:33.557Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":47914583,"total":47914583},"progress":"[==================================================\u003e]  47.91MB/47.91MB","id":"8b4274ea61c5"}
2025-05-30T16:11:33.6372907Z 2025-05-30T16:11:33.635Z testcontainers [INFO] [0aeea7f9867f] Stopping container...
2025-05-30T16:11:33.6373801Z 2025-05-30T16:11:33.635Z testcontainers [DEBUG] [0aeea7f9867f] Stopping container...
2025-05-30T16:11:33.6706421Z 2025-05-30T16:11:33.670Z testcontainers:pull [mysql:8.0.39] {"status":"Pull complete","progressDetail":{},"id":"8b4274ea61c5"}
2025-05-30T16:11:33.6789906Z 2025-05-30T16:11:33.678Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":883,"total":883},"progress":"[==================================================\u003e]     883B/883B","id":"65b13290a890"}
2025-05-30T16:11:33.6869347Z 2025-05-30T16:11:33.686Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":883,"total":883},"progress":"[==================================================\u003e]     883B/883B","id":"65b13290a890"}
2025-05-30T16:11:34.0089106Z 2025-05-30T16:11:34.008Z testcontainers:pull [mysql:8.0.39] {"status":"Pull complete","progressDetail":{},"id":"65b13290a890"}
2025-05-30T16:11:34.0621658Z 2025-05-30T16:11:34.060Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":32768,"total":913441},"progress":"[=\u003e                                                 ]  32.77kB/913.4kB","id":"4a0edbf0dd13"}
2025-05-30T16:11:34.1085866Z PASS __tests__/count-handler.test.ts
2025-05-30T16:11:34.1420306Z 2025-05-30T16:11:34.141Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":913441,"total":913441},"progress":"[==================================================\u003e]  913.4kB/913.4kB","id":"4a0edbf0dd13"}
2025-05-30T16:11:34.1474944Z 2025-05-30T16:11:34.142Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":913441,"total":913441},"progress":"[==================================================\u003e]  913.4kB/913.4kB","id":"4a0edbf0dd13"}
2025-05-30T16:11:34.1511460Z 2025-05-30T16:11:34.150Z testcontainers:pull [mysql:8.0.39] {"status":"Pull complete","progressDetail":{},"id":"4a0edbf0dd13"}
2025-05-30T16:11:34.1639170Z 2025-05-30T16:11:34.163Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":65536,"total":6330567},"progress":"[\u003e                                                  ]  65.54kB/6.331MB","id":"f8d978a6739e"}
2025-05-30T16:11:34.2564712Z PASS __tests__/adapters/driven/CertificateImageService.test.ts
2025-05-30T16:11:34.2649387Z 2025-05-30T16:11:34.264Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":1376256,"total":6330567},"progress":"[==========\u003e                                        ]  1.376MB/6.331MB","id":"f8d978a6739e"}
2025-05-30T16:11:34.3707060Z 2025-05-30T16:11:34.370Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":3080192,"total":6330567},"progress":"[========================\u003e                          ]   3.08MB/6.331MB","id":"f8d978a6739e"}
2025-05-30T16:11:34.3708943Z 2025-05-30T16:11:34.369Z testcontainers [DEBUG] [6f06abd1da49] Connecting to Reaper (attempt 4) on "172.17.0.1:32768"...
2025-05-30T16:11:34.3710350Z 2025-05-30T16:11:34.370Z testcontainers [ERROR] Reaper 6f06abd1da49576a9a98c3ae673165b0881f828cfa719eef3ff57c049472e50e socket error: Error: connect ECONNREFUSED 172.17.0.1:32768
2025-05-30T16:11:34.3711466Z 2025-05-30T16:11:34.370Z testcontainers [ERROR] [6f06abd1da49] Connection to Reaper closed with error
2025-05-30T16:11:34.4234046Z 2025-05-30T16:11:34.423Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":6330567,"total":6330567},"progress":"[==================================================\u003e]  6.331MB/6.331MB","id":"f8d978a6739e"}
2025-05-30T16:11:34.4340563Z 2025-05-30T16:11:34.433Z testcontainers:pull [mysql:8.0.39] {"status":"Pull complete","progressDetail":{},"id":"f8d978a6739e"}
2025-05-30T16:11:34.4359826Z 2025-05-30T16:11:34.435Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":2604,"total":2604},"progress":"[==================================================\u003e]  2.604kB/2.604kB","id":"64da6e9087ad"}
2025-05-30T16:11:34.4361173Z 2025-05-30T16:11:34.435Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":2604,"total":2604},"progress":"[==================================================\u003e]  2.604kB/2.604kB","id":"64da6e9087ad"}
2025-05-30T16:11:34.4834613Z PASS __tests__/ts/StoreBulkDefinition.test.ts
2025-05-30T16:11:34.5163267Z 2025-05-30T16:11:34.515Z testcontainers:pull [mysql:8.0.39] {"status":"Pull complete","progressDetail":{},"id":"64da6e9087ad"}
2025-05-30T16:11:34.5180129Z 2025-05-30T16:11:34.517Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":335,"total":335},"progress":"[==================================================\u003e]     335B/335B","id":"f6d5909e94cd"}
2025-05-30T16:11:34.5303921Z 2025-05-30T16:11:34.528Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":335,"total":335},"progress":"[==================================================\u003e]     335B/335B","id":"f6d5909e94cd"}
2025-05-30T16:11:34.6243490Z 2025-05-30T16:11:34.623Z testcontainers:pull [mysql:8.0.39] {"status":"Pull complete","progressDetail":{},"id":"f6d5909e94cd"}
2025-05-30T16:11:34.7138561Z 2025-05-30T16:11:34.713Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":491520,"total":48048247},"progress":"[\u003e                                                  ]  491.5kB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:34.8159472Z 2025-05-30T16:11:34.814Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":7864320,"total":48048247},"progress":"[========\u003e                                          ]  7.864MB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:34.8933246Z 2025-05-30T16:11:34.892Z testcontainers [INFO] [927936aca11c] Stopping container...
2025-05-30T16:11:34.9088254Z 2025-05-30T16:11:34.893Z testcontainers [DEBUG] [927936aca11c] Stopping container...
2025-05-30T16:11:34.9225335Z 2025-05-30T16:11:34.922Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":13271040,"total":48048247},"progress":"[=============\u003e                                     ]  13.27MB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:35.0241257Z 2025-05-30T16:11:35.023Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":19169280,"total":48048247},"progress":"[===================\u003e                               ]  19.17MB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:35.0810902Z PASS __tests__/certificate-changed-handler.test.ts
2025-05-30T16:11:35.1248908Z 2025-05-30T16:11:35.124Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":28016640,"total":48048247},"progress":"[=============================\u003e                     ]  28.02MB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:35.2216396Z PASS __tests__/adapters/driver/GetCertificatePdfPreSignedUrlHandler.test.ts
2025-05-30T16:11:35.2508481Z 2025-05-30T16:11:35.250Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":35880960,"total":48048247},"progress":"[=====================================\u003e             ]  35.88MB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:35.3565340Z 2025-05-30T16:11:35.356Z testcontainers:pull [mysql:8.0.39] {"status":"Extracting","progressDetail":{"current":39813120,"total":48048247},"progress":"[=========================================\u003e         ]  39.81MB/48.05MB","id":"949fd89ec12a"}
2025-05-30T16:11:35.3780161Z 2025-05-30T16:11:35.371Z testcontainers [DEBUG] [6f06abd1da49] Connecting to Reaper (attempt 5) on "172.17.0.1:32768"...
2025-05-30T16:11:35.3781581Z 2025-05-30T16:11:35.372Z testcontainers [ERROR] Reaper 6f06abd1da49576a9a98c3ae673165b0881f828cfa719eef3ff57c049472e50e socket error: Error: connect ECONNREFUSED 172.17.0.1:32768
2025-05-30T16:11:35.3782833Z 2025-05-30T16:11:35.372Z testcontainers [ERROR] [6f06abd1da49] Connection to Reaper closed with error
2025-05-30T16:11:35.3784082Z 2025-05-30T16:11:35.372Z testcontainers [ERROR] [6f06abd1da49] Failed to connect to Reaper
2025-05-30T16:11:35.3784847Z 2025-05-30T16:11:35.372Z testcontainers [DEBUG] Releasing lock file "/tmp/testcontainers-node.lock"...
2025-05-30T16:11:35.3788589Z 2025-05-30T16:11:35.373Z testcontainers [DEBUG] Released lock file "/tmp/testcontainers-node.lock"
```

**Steps to Reproduce**
1. Monorepo with multiple concurrent tests running
2. Testing with different containers: mysql, dynamodb, localstack...
3. Running concurrently with `pnpm -r ... jest`
4. It doesn't fail every time, but it eventually fails both in MacOS (arm) and Linux (arm)

**Environment Information**
- Operating System: Linux Ubuntu & MacOS (Sonoma+)
- Docker Version: 🤷🏽 
- Node version: 20
- Testcontainers version: 10.16

--- a/packages/testcontainers/src/reaper/reaper.test.ts
+++ b/packages/testcontainers/src/reaper/reaper.test.ts
@@ -53,6 +53,24 @@ describe.sequential("Reaper", { timeout: 120_000 }, () => {
     expect(reaper2.containerId).toBe(reaper.containerId);
   });
 
+  it("should create new reaper container when existing reaper cannot be reached", async () => {
+    const reaper = await getReaper();
+    vi.resetModules();
+    const unreachablePort = await new RandomPortGenerator().generatePort();
+    const reaperContainerInfo = (await client.container.list()).filter((c) => c.Id === reaper.containerId)[0];
+    reaperContainerInfo.Labels["TESTCONTAINERS_RYUK_TEST_LABEL"] = "false";
+    const reaperPort = reaperContainerInfo.Ports.find((port) => port.PrivatePort == 8080);
+    if (!reaperPort) {
+      throw new Error("Expected Reaper to map exposed port 8080");
+    }
+    reaperPort.PublicPort = unreachablePort;
+    vi.spyOn(client.container, "list").mockResolvedValue([reaperContainerInfo]);
+
+    const reaper2 = await getReaper();
+
+    expect(reaper2.containerId).not.toBe(reaper.containerId);
+  });
+
   it("should use custom port when TESTCONTAINERS_RYUK_PORT is set", async () => {
     const customPort = (await new RandomPortGenerator().generatePort()).toString();
     vi.stubEnv("TESTCONTAINERS_RYUK_PORT", customPort);

--- a/packages/testcontainers/src/reaper/reaper.ts
+++ b/packages/testcontainers/src/reaper/reaper.ts
@@ -29,30 +29,44 @@ export async function getReaper(client: ContainerRuntimeClient): Promise<Reaper>
   }
 
   reaper = await withFileLock("testcontainers-node.lock", async () => {
-    const reaperContainer = await findReaperContainer(client);
-    sessionId = reaperContainer?.Labels[LABEL_TESTCONTAINERS_SESSION_ID] ?? new RandomUuid().nextUuid();
+    const reaperContainers = await findReaperContainers(client);
 
     if (process.env.TESTCONTAINERS_RYUK_DISABLED === "true") {
+      sessionId = new RandomUuid().nextUuid();
       return new DisabledReaper(sessionId, "");
-    } else if (reaperContainer) {
-      return await useExistingReaper(reaperContainer, sessionId, client.info.containerRuntime.host);
-    } else {
-      return await createNewReaper(sessionId, client.info.containerRuntime.remoteSocketPath);
     }
+
+    for (const reaperContainer of reaperContainers) {
+      const existingSessionId = reaperContainer.Labels[LABEL_TESTCONTAINERS_SESSION_ID] ?? new RandomUuid().nextUuid();
+      try {
+        sessionId = existingSessionId;
+        return await useExistingReaper(reaperContainer, sessionId, client.info.containerRuntime.host);
+      } catch (error) {
+        const message = error instanceof Error ? error.message : String(error);
+        log.warn(`Failed to reuse existing Reaper: ${message}. Trying another Reaper...`, {
+          containerId: reaperContainer.Id,
+        });
+      }
+    }
+
+    sessionId = new RandomUuid().nextUuid();
+    return await createNewReaper(sessionId, client.info.containerRuntime.remoteSocketPath);
   });
 
   reaper.addSession(sessionId);
   return reaper;
 }
 
-async function findReaperContainer(client: ContainerRuntimeClient): Promise<ContainerInfo | undefined> {
+async function findReaperContainers(client: ContainerRuntimeClient): Promise<ContainerInfo[]> {
   const containers = await client.container.list();
-  return containers.find(
-    (container) =>
-      container.State === "running" &&
-      container.Labels[LABEL_TESTCONTAINERS_RYUK] === "true" &&
-      container.Labels["TESTCONTAINERS_RYUK_TEST_LABEL"] !== "true"
-  );
+  return containers
+    .filter(
+      (container) =>
+        container.State === "running" &&
+        container.Labels[LABEL_TESTCONTAINERS_RYUK] === "true" &&
+        container.Labels["TESTCONTAINERS_RYUK_TEST_LABEL"] !== "true"
+    )
+    .sort((a, b) => b.Created - a.Created);
 }
 
 async function useExistingReaper(reaperContainer: ContainerInfo, sessionId: string, host: string): Promise<Reaper> {

